exasol_advanced_analytics_framework/deployment/scripts_deployer.py

```diff
@@ -49,13 +49,21 @@ def deploy_scripts(self) -> None:
         self._deploy_lua_scripts()
         logger.debug(f"Scripts are deployed.")
 
-    @classmethod
-    def run(cls, dsn: str, user: str, password: str,
-            schema: str, language_alias: str, develop: bool):
+    # @classmethod
+    # def run_old(cls, dsn: str, user: str, password: str,
+    #         schema: str, language_alias: str, develop: bool):
+    #     pyexasol_conn = pyexasol.connect(dsn=dsn, user=user, password=password)
+    #     cls.run2(pyexasol_conn, schema, language_alias, develop)
 
+    @classmethod
```

Comment:
I preferred passing the `pyexasol_connection` instead of the individual parameters `dsn`, `user` and `password` as the tests are using the fixture `pyexasol_connection` anyway and extracting the former individual parameters seemed cumbersome to me.

Reply:
The constructor does not call `save_aaf_query_loop_lua_script()`, nor `scripts_deployer.deploy_scripts()`. 
So in the end I still prefer changing the signature. 
Alternatively we could use a different name and keep the old method.

Reply:
For explanation, the run is specifically for the CLI and for nothing else. Other code should call deploy_scripts directly and should use an already generated Lua script.

Reply:
I found 3 places calling `ScriptsDeployer.run()`:
* P1) [scripts_deployer_cli.py](https://github.com/exasol/advanced-analytics-framework/blob/refactoring/%23174-Replace_Language_Container_Stuff_with_PEC_and_SLC_pytest_plugin/exasol_advanced_analytics_framework/deployment/scripts_deployer_cli.py)
* P2) [tests/deployment/test_scripts_deployer.py](https://github.com/exasol/advanced-analytics-framework/blob/refactoring/%23174-Replace_Language_Container_Stuff_with_PEC_and_SLC_pytest_plugin/tests/deployment/test_scripts_deployer.py)
* P3) [tests/integration_tests/with_db/fixtures/setup_database_fixture.py](https://github.com/exasol/advanced-analytics-framework/blob/refactoring/%23174-Replace_Language_Container_Stuff_with_PEC_and_SLC_pytest_plugin/tests/integration_tests/with_db/fixtures/setup_database_fixture.py)

The current implementation of the [run()](https://github.com/exasol/advanced-analytics-framework/blob/refactoring/%23174-Replace_Language_Container_Stuff_with_PEC_and_SLC_pytest_plugin/exasol_advanced_analytics_framework/deployment/scripts_deployer.py) method is

```python
if develop:
    save_aaf_query_loop_lua_script()
scripts_deployer = cls(language_alias, schema, pyexasol_conn)
scripts_deployer.deploy_scripts()
```

As P2 and P3 both are using flag `develop=True` I will use the following code in order to use the constructor:

```python
save_aaf_query_loop_lua_script()
ScriptsDeployer(language_alias, db_schema_name, pyexasol_connection).deploy_scripts()
```

Reply:
That should be ok for now. We could also think about moving the save_aaf_query_loop_lua_script call into its own fixture